aspnetcore/security/how-to-choose-identity-solution.md

```diff
@@ -0,0 +1,114 @@
+
+The main disadvantage to using tokens is that they require a token server to both issue and provide validation for tokens. The token server must be installed, configured, and maintained.
+
+A common reason a token server is required is for applications that exposes web-based APIs that are consumed by other apps. For exposed web-based APIs, client UIs such as Single Page Applications (SPA), mobile clients, and desktop clients are considered to be part of the same app. SPA examples include Angular, React, and Blazor WebAssembly. If apps other than your web app or any client UIs must make a secure API call to your app, you'll likely want to use tokens. If you only have client UIs, ASP.NET Core Identity provides the option to acquire a token during authentication. The authentication token:
+
+* Can be used by mobile and desktop clients.
+* Isn't suitable for managing access from external apps.
+
+Another reason a token server is required is for sharing sign-ins with other apps. Commonly referred to as *single sign on*, this feature enables users to:
+
+* Sign in once with a web app's form.
+* Use the resulting credentials to authenticate with other apps without having to sign-in again or choose a different password.
+
+A token server is typically preferred to provide a secure and scalable solution for single sign on. 
+
+For apps that don't expose web-based APIs that are consumed by other apps and don't share logins with other apps, the simplest way to quickly secure an app is to use the built-in ASP.NET Core Identity provider. Otherwise, a token server provided by a third-party identity management solution is needed. Token servers are available as:
+
+* Products you install on your server, called *self-host*.
+* Containers run in a host like Docker.
+* Web-based services you integrate with to manage identity.
+
+Some solutions are free and open source, while others are commercially licensed. See [identity management solutions](xref:security/identity-management-solutions) for a list of available options. It is possible that your organization already uses an existing identity provider. In that case, it may make sense to use the existing provider instead of going with a different solution. All of the major solutions provide documentation for configuring ASP.NET Core to use their product or service.
+
+## Disconnected scenarios
+
+Many solutions, such as [Microsoft Entra ID](/azure/active-directory/fundamentals/active-directory-whatis), are cloud-based and require an Internet connection to work. If your environment doesn't allow Internet connectivity, you won't be able to use the service. If you must support scenarios where either your app cannot access the Internet or must still function on the local network even if the Internet is disconnected, choose one of the following options:
+
+* A solution that allows you to install and run the service on your own machines.
+* Run the authentication service locally as a container.
```

Comment:
Shouldn't we be recommending Identity for "disconnected" scenarios as well?

Reply:
> Shouldn't we be recommending Identity for "disconnected" scenarios as well?

Yes. I'll update the first bullet to make this more clear.